### ISSUE TITLE
VSCode Remote C++ Debug not working. 

### ISSUE THREAD

**Issue by liushiqi9** [microsoft/vscode#74655]
## System Version
Version: 1.35.0-insider
Commit: 5c65d9bfa4c56538150d7f3066318e0db2c6151f
Date: 2019-05-30T21:57:25.685Z
Electron: 3.1.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 6.1.7601

## Steps to Reproduce:
1. Download lastest vscode insider with remote extension package and C/C++ extension.
2. Connect to a Arch server with remote ssh.
3. git clone https://github.com/microsoft/vscode-remote-try-cpp.git
4. Set a breakpoint at line 7.
5. Press F5
6. The debug hangs, binary is compiled, but the gdb is not started.  Debug pannel at top-middle apears with step over/into/out button disabled. 

## Logs
![image](https://user-images.githubusercontent.com/6860451/58681033-24157180-839d-11e9-9c46-8d17e28300ed.png)

### Debug console:
```
--> E (output): {"event":"output","body":{"category":"console","output":"1: (259) LaunchOptions<LocalLaunchOptions xmlns='http://schemas.microsoft.com/vstudio/MDDDebuggerOptions/2014'\n"},"seq":2,"type":"event"}
1: (259) LaunchOptions<LocalLaunchOptions xmlns='http://schemas.microsoft.com/vstudio/MDDDebuggerOptions/2014'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (276) LaunchOptions  ExePath='/home/sqliu/workspace/vscode-remote-try-cpp/main.out'\n"},"seq":4,"type":"event"}
1: (276) LaunchOptions  ExePath='/home/sqliu/workspace/vscode-remote-try-cpp/main.out'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (276) LaunchOptions  WorkingDirectory='/home/sqliu/workspace/vscode-remote-try-cpp'\n"},"seq":6,"type":"event"}
1: (276) LaunchOptions  WorkingDirectory='/home/sqliu/workspace/vscode-remote-try-cpp'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (277) LaunchOptions  ExeArguments=''\n"},"seq":8,"type":"event"}
1: (277) LaunchOptions  ExeArguments=''
--> E (output): {"event":"output","body":{"category":"console","output":"1: (277) LaunchOptions  MIMode='gdb'\n"},"seq":10,"type":"event"}
1: (277) LaunchOptions  MIMode='gdb'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (278) LaunchOptions  MIDebuggerPath=''\n"},"seq":12,"type":"event"}
1: (278) LaunchOptions  MIDebuggerPath=''
--> E (output): {"event":"output","body":{"category":"console","output":"1: (279) LaunchOptions  WaitDynamicLibLoad='false'\n"},"seq":14,"type":"event"}
1: (279) LaunchOptions  WaitDynamicLibLoad='false'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (279) LaunchOptions  ExternalConsole='true'\n"},"seq":16,"type":"event"}
1: (279) LaunchOptions  ExternalConsole='true'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (280) LaunchOptions>\n"},"seq":18,"type":"event"}
1: (280) LaunchOptions>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (280) LaunchOptions    <SetupCommands>\n"},"seq":20,"type":"event"}
1: (280) LaunchOptions    <SetupCommands>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (281) LaunchOptions        <Command IgnoreFailures='true' Description='Enable pretty-printing for gdb'>-enable-pretty-printing</Command>\n"},"seq":22,"type":"event"}
1: (281) LaunchOptions        <Command IgnoreFailures='true' Description='Enable pretty-printing for gdb'>-enable-pretty-printing</Command>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (281) LaunchOptions    </SetupCommands>\n"},"seq":24,"type":"event"}
1: (281) LaunchOptions    </SetupCommands>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (282) LaunchOptions</LocalLaunchOptions>\n"},"seq":26,"type":"event"}
1: (282) LaunchOptions</LocalLaunchOptions>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (497) DbgCmd:echo $$ > /tmp/Microsoft-MIEngine-Pid-aj2vlpzw.9t1 ; cd /home/sqliu/workspace/vscode-remote-try-cpp ; DbgTerm=`tty` ; set -o monitor ; trap 'rm /tmp/Microsoft-MIEngine-In-fegd1acm.cim /tmp/Microsoft-MIEngine-Out-4j4hsabw.gch /tmp/Microsoft-MIEngine-Pid-aj2vlpzw.9t1 /tmp/Microsoft-MIEngine-Cmd-1t2icdc8.8cu' EXIT ; /usr/bin/gdb --interpreter=mi --tty=$DbgTerm < /tmp/Microsoft-MIEngine-In-fegd1acm.cim > /tmp/Microsoft-MIEngine-Out-4j4hsabw.gch & clear; pid=$! ; echo $pid > /tmp/Microsoft-MIEngine-Pid-aj2vlpzw.9t1 ; wait $pid; \n"},"seq":28,"type":"event"}
1: (497) DbgCmd:echo $$ > /tmp/Microsoft-MIEngine-Pid-aj2vlpzw.9t1 ; cd /home/sqliu/workspace/vscode-remote-try-cpp ; DbgTerm=`tty` ; set -o monitor ; trap 'rm /tmp/Microsoft-MIEngine-In-fegd1acm.cim /tmp/Microsoft-MIEngine-Out-4j4hsabw.gch /tmp/Microsoft-MIEngine-Pid-aj2vlpzw.9t1 /tmp/Microsoft-MIEngine-Cmd-1t2icdc8.8cu' EXIT ; /usr/bin/gdb --interpreter=mi --tty=$DbgTerm < /tmp/Microsoft-MIEngine-In-fegd1acm.cim > /tmp/Microsoft-MIEngine-Out-4j4hsabw.gch & clear; pid=$! ; echo $pid > /tmp/Microsoft-MIEngine-Pid-aj2vlpzw.9t1 ; wait $pid;
--> C (runInTerminal-30): {"command":"runInTerminal","arguments":{"kind":"external","title":"cppdbg: main.out","cwd":"","args":["sh","/tmp/Microsoft-MIEngine-Cmd-1t2icdc8.8cu"],"env":{}},"seq":30,"type":"request"}
--> E (output): {"event":"output","body":{"category":"console","output":"1: (534) Wait for connection completion.\n"},"seq":32,"type":"event"}
1: (534) Wait for connection completion.

```

## Appearance
![image](https://user-images.githubusercontent.com/6860451/58681071-5b841e00-839d-11e9-8737-4ffed6b109e9.png)

## Expect
Expect the program to hang at the breakpoint with step into/over/out button available. 

All help needed, Thx.

**Comment by liushiqi9** [comment]
#73987
I reported this issue originally at May 20th. It's closed but not fixed.

**Comment by liushiqi9** [comment]
Fixed at Stable 1.35.1
